var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -659,6 +699,63 @@ def cmake_args(self):
 
         # ################# Miscellaneous Stuff ######################
 
+        # CUDA
+        if '+cuda' in spec:
+            options.extend([
+                '-DTPL_ENABLE_CUDA:BOOL=ON',
+                '-DTPL_ENABLE_CUSPARSE:BOOL=ON'
+            ])
+            arch_args = []
+            # Host architectures
+            host_arch = spec.variants['host_arch'].value
+            # GPU architectures
+            gpu_arch  = spec.variants['gpu_arch'].value
+            if host_arch != 'none':
+                arch_args.append(host_arch)
+            if gpu_arch != 'none':
+                arch_args.append(gpu_arch)
+            # Combined architecture flags
+            if arch_args:
+                options.extend([
+                    '-DKOKKOS_ARCH={0}'.format(';'.join(arch_args))
+                ])
+            if '+kokkos_enable_cuda_lambda' in spec:
+                options.extend([
+                    '-DKokkos_ENABLE_Cuda_Lambda:BOOL=ON'
+                ])
+            if '+kokkos_enable_rdc' in spec:
+                options.extend([
+                    '-DKokkos_ENABLE_Cuda_Relocatable_Device_Code:BOOL=ON'
+                ])
+            if '+kokkos_enable_cuda_uvm' in spec:
+                options.extend([
+                    '-DKokkos_ENABLE_Cuda_UVM:BOOL=ON'
+                ])
+            if '+phalanx' in spec:
+                options.extend([
+                    '-DPhalanx_KOKKOS_DEVICE_TYPE:STRING=CUDA'
+                ])
+            if '+sacado' in spec:
+                options.extend([
+                    '-DSacado_ENABLE_HIERARCHICAL_DFAD:BOOL=ON'
+                ])
+            if '+muelu' in spec:
+                options.extend([
+                    '-DMueLu_ENABLE_Kokkos_Refactor_Use_By_Default:BOOL=ON'
+                ])
+            if '+tpetra' in spec:
+                options.extend([
+                    '-DTpetra_INST_CUDA:BOOL=ON',
+                    '-DTpetra_ASSUME_CUDA_AWARE_MPI:BOOL=ON',
```

Comment:
I'm not sure why this is here. CUDA aware MPI is not checked anywhere. Plus, it seems that CUDA aware MPI is currently slower than regular one, and not used in practice.

Reply:
Remove in e022519.